### PR TITLE
[DEV-5734] calc metrics jira groups

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
This change is [Reviewable](https://reviewable.io/reviews/athenianco/api-spec/153)

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -8242,6 +8242,14 @@ components:
           description: Issues must be created before this date.
           format: date
           type: string
+        groups:
+          description: |
+            Calculate metrics separately for each JIRA Filter in the list.
+            Not compatible with other jira filters in the request:
```
Comment (gaetano-guerriero) [inline openapi.yaml:8248]: Note that this limitation is not expressed by the schema: our typescript generator would explode with not/anyOf or allOf/not combinations. The API will check this condition manually.

File: openapi.yaml [hunk 2]
```diff
@@ -8242,6 +8242,14 @@ components:
           description: Issues must be created before this date.
           format: date
           type: string
+        groups:
```
Comment (gaetano-guerriero) [inline openapi.yaml:8245]: @vmarkovtsev suggestion for a better name is well accepted

**Review by akbarik — APPROVED** [review]

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/api-spec/153)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @gaetano-guerriero and @vmarkovtsev)

___
*[`openapi.yaml` line 8245 at r1](https://reviewable.io/reviews/athenianco/api-spec/153#-NO9PViiONEhCGcHLAwu-r1-8245:-NO9Plq4AYIkbmo0HNev:b-j8rw3h) ([raw file](https://github.com/athenianco/api-spec/blob/43835b56954683e78525349bd84402b7c99ad5bf/openapi.yaml#L8245)):*
<details><summary><i>Previously, gaetano-guerriero wrote…</i></summary><blockquote>

@vmarkovtsev suggestion for a better name is well accepted
</blockquote></details>

We seem to stick to "for" in the other endpoints, so let's use "for".
___
*[`openapi.yaml` line 8248 at r1](https://reviewable.io/reviews/athenianco/api-spec/153#-NO9PViiONEhCGcHLAwu-r1-8248:-NO9Q0oF6zJpRFxNo2bP:b5wtvic) ([raw file](https://github.com/athenianco/api-spec/blob/43835b56954683e78525349bd84402b7c99ad5bf/openapi.yaml#L8248)):*
<details><summary><i>Previously, gaetano-guerriero wrote…</i></summary><blockquote>

Note that this limitation is not expressed by the schema: our typescript generator would explode with not/anyOf or allOf/not combinations. The API will check this condition manually.
</blockquote></details>

We would ideally rewrite this endpoint usage in the frontend to always use `for`/`groups` to stay uniform with the rest.
BTW the reason why the JIRA metrics doesn't allow grouping is just that it required more code and we were in quite a hurry to add JIRA asap.

**Review by gaetano-guerriero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/api-spec/153)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @vmarkovtsev)

___
*[`openapi.yaml` line 8248 at r1](https://reviewable.io/reviews/athenianco/api-spec/153#-NO9PViiONEhCGcHLAwu-r1-8248:-NO9Wfy00YjDrgHWATwG:b-qee4kv) ([raw file](https://github.com/athenianco/api-spec/blob/43835b56954683e78525349bd84402b7c99ad5bf/openapi.yaml#L8248)):*
<details><summary><i>Previously, vmarkovtsev (Vadim Markovtsev) wrote…</i></summary><blockquote>

We would ideally rewrite this endpoint usage in the frontend to always use `for`/`groups` to stay uniform with the rest.
BTW the reason why the JIRA metrics doesn't allow grouping is just that it required more code and we were in quite a hurry to add JIRA asap.
</blockquote></details>

We can start migrating the calls to `for`, I guess we  have to keep the rest of parameters to not break compatibility ?

**Review by vmarkovtsev — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/api-spec/153)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @vmarkovtsev)

___
*[`openapi.yaml` line 8248 at r1](https://reviewable.io/reviews/athenianco/api-spec/153#-NO9PViiONEhCGcHLAwu-r1-8248:-NOAIgrI1q978zz64xhk:b-pzwto5) ([raw file](https://github.com/athenianco/api-spec/blob/43835b56954683e78525349bd84402b7c99ad5bf/openapi.yaml#L8248)):*
<details><summary><i>Previously, gaetano-guerriero wrote…</i></summary><blockquote>

We can start migrating the calls to `for`, I guess we  have to keep the rest of parameters to not break compatibility ?
</blockquote></details>

Yep, backward-compatible

**Pull request merged**